lib/galaxy/webapps/galaxy/api/job_tokens.py

```diff
@@ -0,0 +1,78 @@
+""" API for asynchronous job running mechanisms can use to fetch or put files
+related to running and queued jobs.
+"""
+import logging
+
+from galaxy import (
+    exceptions,
+    model,
+    util,
+)
+from galaxy.authnz.util import provider_name_to_backend
+from galaxy.web import expose_api_raw_anonymous_and_sessionless
+from . import BaseGalaxyAPIController
+
+log = logging.getLogger(__name__)
+
+
+class JobTokensAPIController(BaseGalaxyAPIController):
+    """This job token controller allows remote job running mechanisms to
+    get a fresh OIDC token that can be used on remote side to authorize user.
+    It is not meant to represent part of Galaxy's stable, user
+    facing API.
+    """
+
+    @expose_api_raw_anonymous_and_sessionless
+    def get_token(self, trans, job_id, **kwargs):
+        """
+        GET /api/jobs/{job_id}/oidc-tokens
+
+        Get a fresh OIDC token
+
+        :type   job_id: str
+        :param  job_id: encoded id string of the job
+        :type   provider: str
+        :param  provider: Path to file.
+        :type   job_key: str
+        :param  job_key: A key used to authenticate this request as acting on
+                         behalf or a job runner for the specified job.
+
+        ..note:
+            This API method is intended only for consumption by job runners,
+            not end users.
+
+        :rtype:     str
+        :returns:   OIDC ID token
+        """
+        job = self.__authorize_job_access(trans, job_id, **kwargs)
+        trans.app.authnz_manager.refresh_expiring_oidc_tokens(trans, job.user)  # type: ignore[attr-defined]
+        provider = kwargs.get("provider", None)
+        tokens = job.user.get_oidc_tokens(provider_name_to_backend(provider))
+        return tokens["id"]
+
+    def __authorize_job_access(self, trans, encoded_job_id, **kwargs):
+        for key in ["provider", "job_key"]:
+            if key not in kwargs:
+                error_message = f"Job files action requires a valid '{key}'."
+                raise exceptions.ObjectAttributeMissingException(error_message)
+
+        job_id = trans.security.decode_id(encoded_job_id)
+        job = trans.sa_session.query(model.Job).get(job_id)
+        secret = "jobs_token"
+        try:
+            for plugin in self.app.job_config.runner_plugins:
+                if plugin["id"] == job.job_runner_name:
+                    secret = plugin["kwds"]["secret"]
```

Comment:
This should probably operate on the destination instead of the job runner plugin level, since this doesn't really seem tied to the job runner (even if it only works for pulsar) and more tied to e.g. the user.
Also could `secret` be more descriptive ? maybe `job_oidc_token_secret_base` ?

Reply:
Makes sense, thank you. Just one comment -  this secret can be used in other places for calling Galaxy API from some service. I.e. to get. job files. Currently that secret is hard-coded in the [source](https://github.com/galaxyproject/galaxy/blob/2be2cec8c5bee607cf04fc279f173af82d5f3627/lib/galaxy/webapps/galaxy/api/job_files.py#L124), which is not so nice. So I suggest to use `job_secret_base` instead of `job_oidc_token_secret_base `